src/components/dialog/dialog.ts

```diff
@@ -0,0 +1,113 @@
+import {Injector, ComponentRef, Injectable} from '@angular/core';
+import {Overlay} from '@angular2-material/core/overlay/overlay';
+import {OverlayRef} from '@angular2-material/core/overlay/overlay-ref';
+import {OverlayState} from '@angular2-material/core/overlay/overlay-state';
+import {ComponentPortal} from '@angular2-material/core/portal/portal';
+import {ComponentType} from '@angular2-material/core/overlay/generic-component-type';
+import {MdDialogConfig} from './dialog-config';
+import {MdDialogRef} from './dialog-ref';
+import {DialogInjector} from './dialog-injector';
+import {MdDialogContainer} from './dialog-container';
+
+
+export {MdDialogConfig} from './dialog-config';
+export {MdDialogRef} from './dialog-ref';
+
+
+// TODO(jelbourn): add shortcuts for `alert` and `confirm`.
+// TODO(jelbourn): add support for opening with a TemplateRef
+// TODO(jelbourn): add `closeAll` method
+// TODO(jelbourn): add backdrop
+// TODO(jelbourn): default dialog config
+// TODO(jelbourn): focus trapping
+// TODO(jelbourn): potentially change API from accepting component constructor to component factory.
+
+
+
+/**
+ * Service to open Material Design modal dialogs.
+ */
+@Injectable()
+export class MdDialog {
```

Comment:
This is in `components/...` but is not a component. As a general discussion should we have a naming convention for services that aren't in core? Is there going to be an `MdDialog` that's actually a component?